.github/workflows/go-test.yaml

```diff
@@ -0,0 +1,39 @@
+name: Plugin Go test
+on:
+  pull_request:
+
+defaults:
+  run:
+    # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+jobs:
+  with-krew:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: setup golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.19
+
+      - name: setup cluster
+        uses: ./.github/workflows/cluster-setup
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: build the binary
+        run: |
+          go build -o kubectl-rook-ceph cmd/main.go
```

Comment:
Instead of running go build in this action, we need a makefile. Then developers can easily and consistently build locally during development, and the CI can also call `make`. But let's try to make it much simpler than the rook makefile. :)

Reply:
created makefile, I think this is one of the simplest makefile you will come across 😅

Reply:
Simple makefiles are the best kind!